src/test/regress/sql/pg16.sql

```diff
@@ -650,10 +638,142 @@ RESET citus.enable_ddl_propagation;
+GRANT role11 TO role12 WITH INHERIT FALSE, ADMIN TRUE, SET FALSE;
+-- Grant role with inherit set to false
+GRANT role13 TO role14 WITH INHERIT FALSE;
+-- Grant role with set option set to false
+GRANT role17 TO role18 WITH SET FALSE;
+DROP ROLE role17, role18;
+-- Handles with default inherit false
+GRANT role19 TO role20;
+
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+\c - - - :worker_1_port
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+\c - - - :master_port
+REVOKE role6 FROM role7;
+
+SELECT roleid::regrole::text AS role, member::regrole::text,
+admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+\c - - - :worker_1_port
+SELECT roleid::regrole::text AS role, member::regrole::text,
+admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role6' ORDER BY 1, 2;
+\c - - - :master_port
+DROP ROLE role6, role7;
+
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role9' ORDER BY 1, 2;
+\c - - - :worker_1_port
+SELECT roleid::regrole::text AS role, member::regrole::text,
+admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role9' ORDER BY 1, 2;
+\c - - - :master_port
+DROP ROLE role9, role10;
+
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role11' ORDER BY 1, 2;
+\c - - - :worker_1_port
+SELECT roleid::regrole::text AS role, member::regrole::text,
+admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role11' ORDER BY 1, 2;
+\c - - - :master_port
+DROP ROLE role11, role12;
+
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role13' ORDER BY 1, 2;
+\c - - - :worker_1_port
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role13' ORDER BY 1, 2;
+\c - - - :master_port
+DROP ROLE role13, role14;
+
+SELECT roleid::regrole::text AS role, member::regrole::text, admin_option, inherit_option,set_option FROM pg_auth_members
+WHERE roleid::regrole::text = 'role19' ORDER BY 1, 2;
+\c - - - :worker_1_port
+SELECT roleid::regrole::text AS role, member::regrole::text,
+admin_option, inherit_option, set_option
+FROM pg_auth_members
+WHERE roleid::regrole::text = 'role19' ORDER BY 1, 2;
+\c - - - :master_port
+DROP ROLE role19, role20;
+
+\c - - - :master_port
```

Comment:
Let's drop all the roles here before starting the next test
```suggestion
\c - - - :master_port
DROP ROLE role6, role7, role8, role9, role10, role11, role12,
          role13, role14, role15, role16, role17, role18;
```